docs/decisions/app/0005-server-rendering.md

```diff
@@ -0,0 +1,67 @@
+- Makes it easier to implement a clearer separation of concerns:
+  - [Middleware](https://nextjs.org/docs/advanced-features/middleware&sa=D&source=editors&ust=1675374865787476&usg=AOvVaw3zDFoDM50pSoew_4SsZpjm) provide a place for enforcing auth, reading/setting secure cookies, setting HTTP headers, and redirects.
+  - [Loaders](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props&sa=D&source=editors&ust=1675374865787877&usg=AOvVaw0zAh7tsjHKl2P8pDGVhf-0) provide a place for fetching all data required for rendering the page.
+- Running page logic and rendering on the server makes it possible to send lighter payloads to the client. This approach can work well for a large spectrum of device and network conditions. [You can make your server fast, but you can't control the user's device or network](https://remix.run/docs/en/v1/pages/philosophy%23serverclient-model&sa=D&source=editors&ust=1675374865788271&usg=AOvVaw3bPMcx9gLfFygQu_qL2R-9).
+- Low effort to implement pretty [dynamic routes](https://nextjs.org/docs/routing/dynamic-routes&sa=D&source=editors&ust=1675374865788677&usg=AOvVaw070KKTAMkbV2YkqYVRpep-) (e.g `/claim/:claim_id`)
+- [API routes](https://nextjs.org/docs/api-routes/introduction) can be created to handle other types of HTTP requests (POST, PUT, etc).
+- Nice side benefit: Server rendering is the only option for [Remix](https://remix.run/), so it may be easier to migrate to Remix if the Next.js apps we're building were server rendered (at least from an infra standpoint?).
+
+Cons
+
+- Requires infra resources to run the containerized application, and all the things that come along with a server (rate limiting, auto scaling), such as [AWS App Runner](https://aws.amazon.com/apprunner/). This can have higher costs than a prerendered site, and requires more effort to receive security approvals due to a larger attack surface.
+
+### Prerendering
+
+Pros
+
+- Great for non-authenticated sites since the markup can be generated ahead of time.
+- Minimal infrastructure is required. The prerendered HTML files can be served from a CDN, such as AWS CloudFront connected to an S3 bucket. Assuming other best practices are followed, like optimizing images and not loading MB's of client-side JS, this can translate to fast page loads and low costs.
+
+Cons
+
+- For sites with live/personalized data, pages would require client-side JS for data fetching. This has a few downsides:
+  - Client-side JS is required for rendering the loading, success, and error states (e.g. `fetch`, `isLoading`, `useEffect`, `catch`). Teams need to define their own code patterns to manage this (e.g hooks, higher-order components) or install third-party dependencies (e.g. [React Query](https://react-query-v3.tanstack.com/)). This increases the amount of code to be written and maintained, and can increase code complexity.
```

Comment:
Something like this? I found the complexity led to so many possible negative edge cases that needed to be handled.
```suggestion
  - Client-side JS is required for rendering the loading, success, and error states (e.g. `fetch`, `isLoading`, `useEffect`, `catch`). Teams need to define their own code patterns to manage this (e.g hooks, higher-order components) or install third-party dependencies (e.g. [React Query](https://react-query-v3.tanstack.com/)). This increases the amount of code to be written and maintained, and can increase code complexity and introduce more opportunity for errors to creep in.
```